compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -88,13 +91,22 @@ object Completion {
           completionPrefix(selector :: Nil, pos)
         }.getOrElse("")
 
+      // We special case Select here because we want to determine if the name
```

Comment:
Yea I'm actually working with figuring this out for `Ident` as well. However one difference I noticed that I don't fully get is that the `name` of a `Select` is a nospan when it's an `<error>`, however the name of an `Ident` when it's an `<error>` has a span. This makes getting the prefix different for each since here in the Select it's a bit hacky. How come one `<error>` is a no span and the other isn't?